frappe/custom/doctype/custom_field/custom_field.py

```diff
@@ -34,6 +46,9 @@ def set_fieldname(self):
 		# fieldnames should be lowercase
 		self.fieldname = self.fieldname.lower()
 
+		if self.fieldname in restricted:
+			self.fieldname = self.fieldname + "1"
```

Comment:
Lets throw error instead of renaming field? :thinking: 

On save it immediately inserts the new columns :pensive:

Reply:
@ankush We cannot set fieldname in customize form (don't know why) fieldname is read-only. Fieldname is generated based on label

Reply:
@shariquerik Label is more user-friendly than fieldname which is why only Label is editable. But in this case we should ask the user that the fieldname is going to be renamed before implicitly setting? This problem can be solved on the client side by using better error messages.